### PR TITLE
8341191: Open source few more AWT FileDialog tests

### PR DESCRIPTION
**Pull request by kumarabhi006**
AWT File Dialog related tests are converted from applet to manual and moved to open.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8341191](https://bugs.openjdk.org/browse/JDK-8341191): Open source few more AWT FileDialog tests (**Bug** - P4)

### Reviewers
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) Review applies to [872bae22](https://git.openjdk.org/jdk/pull/21277/files/872bae2297f7387b91c6bdc0a40c88f192866a75)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21277/head:pull/21277` \
`$ git checkout pull/21277`

Update a local copy of the PR: \
`$ git checkout pull/21277` \
`$ git pull https://git.openjdk.org/jdk.git pull/21277/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21277`

View PR using the GUI difftool: \
`$ git pr show -t 21277`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21277.diff">https://git.openjdk.org/jdk/pull/21277.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21277#issuecomment-2384803279)

### CODE REVIEW

File: test/jdk/java/awt/FileDialog/PathChoiceDisposeTest.java [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+ * @summary Test that disposing unfurled list by the pressing ESC
+ *          in FileDialog is working properly on XToolkit
+ * @library /java/awt/regtesthelpers
+ * @build PassFailJFrame
+ * @run main/manual PathChoiceDisposeTest
+ */
+
+public class PathChoiceDisposeTest {
+    public static void main(String[] args) throws Exception {
+        System.setProperty("sun.awt.disableGtkFileDialogs","true");
```
Comment (turbanoff) [inline test/jdk/java/awt/FileDialog/PathChoiceDisposeTest.java:43]:
```suggestion
        System.setProperty("sun.awt.disableGtkFileDialogs", "true");
```

File: test/jdk/java/awt/FileDialog/SavedDirInitTest.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+ */
+
+import java.awt.Button;
+import java.awt.FileDialog;
+import java.awt.Frame;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+import java.awt.Component;
```
Comment (prsadhuk) [inline test/jdk/java/awt/FileDialog/SavedDirInitTest.java:31]: Component should be after Button..

File: test/jdk/java/awt/FileDialog/SavedDirInitTest.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+            public void actionPerformed(ActionEvent e) {
+                fd.setVisible(true);
+                if (fd.getDirectory() == null && fd.getFile() == null) {
+                    PassFailJFrame.log("TEST PASSED");
+                } else {
+                    PassFailJFrame.log("TEST FAILED. dir = " + fd.getDirectory()
+                            + " , file = " + fd.getFile());
+                }
+                System.out.println("fd components count: "+fd.getComponentCount());
+                System.out.println("fd components: "+fd.getComponents());
```
Comment (prsadhuk) [inline test/jdk/java/awt/FileDialog/SavedDirInitTest.java:76]: guess these also could be put in PassFailJFrame.log

**Review by prsadhuk — APPROVED** [review]

File: test/jdk/java/awt/FileDialog/SavedDirInitTest.java (hunk 2 above)
Comment (kumarabhi006) [reply test/jdk/java/awt/FileDialog/SavedDirInitTest.java]: Actually I was trying to automate the test and import was added, forgot to remove them after that.

Removed the import now.

**Comment by kumarabhi006** [comment]
@prsadhuk Please re-review.

**Review by prrace — APPROVED** [review]

**Pull request closed without merging**